front/scrub_workspace/temporal/activities.ts

```diff
@@ -0,0 +1,125 @@
+  getAgentConfigurations,
+} from "@app/lib/api/assistant/configuration";
+import { isGlobalAgentId } from "@app/lib/api/assistant/global_agents";
+import { deleteDataSource, getDataSources } from "@app/lib/api/data_sources";
+import { getMembers } from "@app/lib/api/workspace";
+import { Authenticator } from "@app/lib/auth";
+import { sendAdminDataDeletionEmail } from "@app/lib/email";
+import {
+  AgentMessage,
+  Conversation,
+  Message,
+  UserMessage,
+} from "@app/lib/models";
+import logger from "@app/logger/logger";
+
+export async function sendDataDeletionEmail({
+  remainingDays,
+  workspaceId,
+}: {
+  remainingDays: number;
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalAdminForWorkspace(workspaceId);
+  const ws = auth.workspace();
+  if (!ws) {
+    throw new Error("No workspace found");
+  }
+  const admins = await getMembers(auth, { roles: ["admin"] });
+  for (const a of admins)
+    await sendAdminDataDeletionEmail({
+      email: a.email,
+      firstName: a.firstName,
+      workspaceName: ws.name,
+      remainingDays,
+    });
+}
+
+export async function shouldStillScrubData({
+  workspaceId,
+}: {
+  workspaceId: string;
+}): Promise<boolean> {
+  return !(
+    await Authenticator.internalAdminForWorkspace(workspaceId)
+  ).isUpgraded();
+}
+
+export async function scrubWorkspaceData({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalAdminForWorkspace(workspaceId);
+  await deleteAllConversations(auth);
+  await archiveAssistants(auth);
+  await deleteDatasources(auth);
+}
+
+async function deleteAllConversations(auth: Authenticator) {
+  const workspace = auth.workspace();
+  if (!workspace) {
+    throw new Error("No workspace found");
+  }
+  const conversations = await Conversation.findAll({
+    where: { workspaceId: workspace.id },
+  });
+  const chunks = chunk(conversations, 4);
+  for (const chunk of chunks) {
+    await Promise.all(
+      chunk.map(async (c) => {
+        logger.info({ conversationId: c.id }, "Deleting conversation");
+        const messages = await Message.findAll({
+          attributes: ["id", "userMessageId", "agentMessageId"],
+          where: { conversationId: c.id },
+        });
+        const userMessageIds = removeNulls(
+          messages.map((m) => m.userMessageId)
+        );
+        const agentMessageIds = removeNulls(
+          messages.map((m) => m.agentMessageId)
+        );
+        await Message.destroy({
+          where: { conversationId: c.id },
+        });
+        await UserMessage.destroy({
+          where: { id: userMessageIds },
+        });
+        await AgentMessage.destroy({
+          where: { id: agentMessageIds },
+        });
+        await c.destroy();
+      })
+    );
+  }
+}
+
+async function archiveAssistants(auth: Authenticator) {
+  const agentConfigurations = await getAgentConfigurations({
+    auth,
+    agentsGetView: "admin_internal",
+    variant: "full",
```

Comment:
if we only need the sid, the light variant should be fine right? (less load)

Reply:
fair